src/karmabot/commands/note.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+import re
+from typing import Callable, Dict, Union
+
+from karmabot.db import db_session
+from karmabot.db.karma_note import KarmaNote
+from karmabot.db.karma_user import KarmaUser
+
+NOTE_CMD_PATTERN = re.compile(r"note\s(\w+)\s?(.*)")
+NOTE_COMMANDS: Dict[str, Callable] = {
+    "add": lambda text_arg, user: _add_note(text_arg, user),
+    "del": lambda text_arg, user: _del_note(text_arg, user),
+    "list": lambda text_arg, user: _list_notes(text_arg, user),
+    "default": lambda text_arg, user: f"Sorry {user.username}, your note command was not recognized. You can use {', '.join(NOTE_COMMANDS.keys())}.",
+}
+
+
+def note(**kwargs) -> Union[None, str]:

```

Comment:
Why do we need `kwargs` here? Why not just pass in `user_id` and `text` explicitly here?
This makes it less transparent what should go into this function.

Reply:
I am with @bbelderbos on this. Rest LGTM

Reply:
Ok so I changed it ;)